### PR TITLE
Fix `or-later`/`only` processing

### LINKED ISSUE #12: Inconsistent behavior with "+" and "-or-later" modifiers.
**Description:**

Although there appears to have been a PR to add support for this, `satisfies` does not seem to handle the `+` modifier and its synonymous `-or-later` modifier consistently:

**Versions Used:**

```
spdx-satisfies: 5.0.0
spdx-correct: 3.1.0
```

**Expected Behavior:**

```js
satisfies('GPL-2.0', 'GPL-2.0+') //=> true
satisfies('GPL-2.0', 'GPL-2.0-or-later') //=> true
```

**Actual Behavior:**

```js
satisfies('GPL-2.0', 'GPL-2.0+') //=> true
satisfies('GPL-2.0', 'GPL-2.0-or-later') //=> false
```

I considered applying `spdx-correct` to both terms before passing them to `satisfies`, but `spdx-correct` normalizes `GPL-2.0+` to `GPL-2.0-or-later`, the case for which `satisfies` produces unexpected results.

---

P.S. Thank you for all the work you've done on the various `spdx-` packages, @kemitchell. 💯

### PR DESCRIPTION
**Pull request by brettz9**
Fixes #12

### CODE REVIEW

**Comment by brettz9** [comment]
As far as Travis failing, it appears the use of `const`, etc. is at issue given that Travis is testing very old Node versions. Not sure if you want to update the versions it is testing (or refactor to restore support for older Node, but note the problem predates this PR, and it appears dependencies have this issue too).

**Comment by kemitchell** [comment]
I'll replace Travis with a GitHub action shortly.

Let's please do stick to ES5. A number of uses run tools that depend on this package on pretty old versions of Node.

**Comment by brettz9** [comment]
Ok, but as mentioned, this was already broken before this PR (so would need to know if you want such changes in this PR).

With the GitHub Action, note that at least the devDependencies this package is using also use `let` (at least `standard` -> `standard-engine` -> `deglob` -> `run-parallel`), so CI will fail if the versions tested are too low.

**Comment by kemitchell** [comment]
I'll handle CI.

**Comment by brettz9** [comment]
Without a Babel step, other items will need refactoring too, such as `Array.from` and `Object.assign` which are in use as well.

File: index.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ function normalizeGPLIdentifiers (argument) {
 }
 
 function endsWith (string, substring) {
-  return string.indexOf(substring) === string.length - 1
+  return string.indexOf(substring) === string.length - substring.length
```
Comment (kemitchell) [inline index.js:79]: So this functionality was broken due to a logic error here?

Comment (brettz9) [reply index.js:79]: Right.

Comment (kemitchell) [reply index.js:79]: Thanks so much for catching and correcting!

**Comment by kemitchell** [comment]
v5.0.1 published

File: index.js (hunk 1 above)
Comment (brettz9) [reply index.js:79]: 🙏 . Thanks for the quick review!

**Pull request merged**